src/api/app/controllers/webui/projects/pulse_controller.rb

```diff
@@ -2,13 +2,18 @@ module Webui
   module Projects
     class PulseController < WebuiController
       before_action :set_project
+      before_action :set_pulse
 
       def show
         switch_to_webui2
         @pulse = @project.project_log_entries.page(params[:page])
       end
 
-      def update_pulse
+      private
+
+      def set_pulse
+        return true unless request.xhr?
```

Comment:
You could do:

```
before_action :set_pulse, only: :xhr?

def xhr?
  request.xhr?
end
```

Maybe the helper method is not necessary but not sure about that

Reply:
Nope, see [_normalize_callback_options](https://api.rubyonrails.org/classes/AbstractController/Callbacks/ClassMethods.html#method-i-_normalize_callback_options)...

Reply:
there is `if: -> {...}` though...